app/src/main/java/com/gruppo4/ringUp/permissions/PermissionsHandler.java

```diff
@@ -55,11 +62,12 @@ public static boolean checkPermissions(Context context, @NonNull String[] permis
     /**
      * Checks if a single permission is granted.
      *
-     * @param permission The permission to check.
+     * @param context    The target Context. It can't be null.
+     * @param permission The permission to check. It can't be null.
      * @return true if the permission is granted, false otherwise.
      * @author Francesco Bau'
      */
-    private static boolean isGranted(Context context, String permission) {
+    private static boolean isGranted(@NonNull Context context, @NonNull String permission) {
```

Comment:
This method is called only inside this class, in particular after you already check if `String[] permissions` is null or not.
For example in line 53 and 32.
After saying that, you can be sure that this variable can NEVER be null.

Reply:
I see.
Thank you for the suggestion.

Reply:
Thank you for the suggestion.
But I've put a @ NonNull annotation because, if this class updates, and another method tries to pass a null String, isGranted would throw an exception, because of method checkSelfPermission(@ NonNull String), and it would be hard to understand immediately, or to test it out, since it's a private method...

I thought this way. Am I wrong? What do you think about it?

Reply:
In doubt, I'm gonna change something... You'll see it in next commit.

Reply:
I changed it, now it's called `isNotGranted` . What do you think about it?